src/psyclone/dynamo0p3.py

```diff
@@ -5460,13 +6492,52 @@ def gen_stub(self):
         # create the subroutine
         sub_stub = SubroutineGen(psy_module, name=base_name+"_code",
                                  implicitnone=True)
-        # create the arglist and declarations
-        create_arg_list = KernStubArgList(self, sub_stub)
+        sub_stub.add(UseGen(sub_stub, name="constants_mod", only=True,
+                            funcnames=["r_def"]))
+
+        iter_cell = DynCellIterators(self)
```

Comment:
Could you iterate over instances here?

```
for xxx in [DynCellIterators, ...]:
    yyy = xxx(self)
    yyy.declarations(sub_stub)
```
with xxx and yyy having some suitable name that I can't think of :-)

Reply:
Nice. I'd thought about doing that but had then forgotten again.